src/workload.rs

```diff
@@ -676,6 +824,19 @@ impl WorkloadStore {
         self.workloads.insert(wip, w);
     }
 
+    fn insert_svc(&mut self, svc: Service) {
+        // if svc already exists, just add new endpoints to existing svc
```

Comment:
this never removes endpoints

Reply:
Actually, looking closer maybe this is bug-free since its only called with Services that have endpoints from staged_vips, but it feels very fragile/confusing

Reply:
I think this is addressed by https://github.com/istio/ztunnel/pull/486/commits/9a0b67b8de69a9302619da99f300bdff0df16555 which moves this mutation to the shared implementation for insert_svc and https://github.com/istio/ztunnel/pull/486/commits/a08e65d78de4d32530baba79c0608c404b6073d4 which fixes the bug for removing endpoints and tests it